pkg/monitor/event.go

```diff
@@ -64,11 +74,16 @@ func startEventMonitoring(ctx context.Context, m Recorder, client kubernetes.Int
 							if !ok {
 								continue
 							}
+							// Record the observed rv version, re-used if we lose our watch connection and re-establish
+							// to prevent missed events.
+							rv = obj.ResourceVersion
```

Comment:
this appears to be missing handling for bookmark events that are sent when a kube-apiserver shuts down.